### DISCUSSION TITLE
easy way to access state via data source and repeater or table

### DISCUSSION THREAD

**Discussion by yeoldegrove** [Budibase/budibase#10447] [Help] [answered]
I have a large table where executing queries takes quite a while (sometimes 60s +). In addition to that I have multiple queries running on a single screen.
My idea was to build some kind of caching via "on screen load -> execute query -> save to state" and later access the state via a data source, repeater or table.
Has anybody ever done such a thing or can think of a way? I cannot find an easy way to access state data via a data source.

**Comment by melohagan** [comment, marked as answer]
Hey @yeoldegrove 

You could do that, and then reference the values of state through bindings. Unfortunately you cannot use state as a source for a Data provider, so you don't get the use of Tables, Repeaters, etc.

Related issues:
 - https://github.com/Budibase/budibase/issues/5917
 - https://github.com/Budibase/budibase/issues/10290

**Comment by yeoldegrove** [comment]
I found a hacky way of accessing state through querying a dummy data source (e.g. bbproxy) which has state as input+output (via transformer).
"works for me" ... But after thinking a bit, using the already existing redis instance as a cache might be the better solution.

**Reply by AlejandroFNadal** [reply]
Can you elaborate on how you did that @yeoldegrove ? Being able to put a state as a dummy data source with a state would be amazing.

**Reply by yeoldegrove** [reply]
Just add a new data source, querying any dummy url (I used, e.g. http:///proxy-service:10000 in my docker-compose setup). Add a binding called `input` and a transformer which just replays your input via `return params.input`. Send the value of the state there by using this dummy data source and it will be mirrored back to you. Only difference is that you can now use it in repeaters/tables.
Quite a hack... I would not use this in production.
The caching via redis is much more sane ... And also reaaaalyyyy brings a performance boost (querying it is much faster that querying the internal budibase db) if you do not rely on too up to date data.
